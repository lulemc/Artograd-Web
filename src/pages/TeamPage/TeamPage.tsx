export const TeamPage = () => {
  return <div>Team Page</div>;
};
