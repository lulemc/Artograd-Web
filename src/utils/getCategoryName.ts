import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPages.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPages.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPages.newTender.categories.murals' },
  { id: 3, name: 'tendersPages.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPages.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPages.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPages.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPages.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPages.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPages.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
