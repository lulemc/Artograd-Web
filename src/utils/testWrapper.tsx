import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { RootState, rootReducer, store } from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { createStore } from '@reduxjs/toolkit';
import { initialState as identityState } from '../store/identitySlice';
import { initialState as helpersState } from '../store/helpersSlice';
import { initialState as profileInformationReducer } from '../store/slices/profileInformationSlice';
import { initialState as profileFundrasingReducer } from '../store/slices/profileFundrasingSlice';
import { PersistPartial } from 'redux-persist/lib/persistReducer';

export const initialState: RootState & PersistPartial = {
  identity: identityState,
  helpers: helpersState,
  profileInformation: profileInformationReducer,
  profileFundrasing: profileFundrasingReducer,
  _persist: {
    version: 1,
    rehydrated: true,
  },
};

export const testWrapper = async ({
  component,
  history,
  path,
  state = initialState,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  path?: string;
  state?: RootState;
}) => {
  return path ? (
    <Provider store={store}>
      <MemoryRouter initialEntries={[path]}>{component}</MemoryRouter>
    </Provider>
  ) : (
    renderWithContextAsync(
      <Provider store={createStore(rootReducer, state)}>
        <Router history={history}>{component}</Router>
      </Provider>,
    )
  );
};
