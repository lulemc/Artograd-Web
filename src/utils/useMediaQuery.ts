import { useEffect, useState } from 'react';

export enum DeviceType {
  mobile = '(max-width: 767px)',
  tablet = '(max-width: 1023px)',
  laptop = '(min-width: 1024px)',
}

export const useMediaQuery = (query: DeviceType) => {
  const mediaMatch = window.matchMedia(query);
  const [matches, setMatches] = useState(mediaMatch.matches);
  useEffect(() => {
    const handler = (e: { matches: boolean }) => setMatches(e.matches);
    mediaMatch.addListener(handler);
    return () => mediaMatch.removeListener(handler);
  });
  return matches;
};
