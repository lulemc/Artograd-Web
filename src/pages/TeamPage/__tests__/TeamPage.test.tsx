import { testWrapper } from '../../../utils/testWrapper';
import { TeamPage } from '../TeamPage';
import { createMemoryHistory } from 'history';

describe('Team page', () => {
  const history = createMemoryHistory();
  test('Should create the team page', async () => {
    const component = await testWrapper({
      component: <TeamPage />,
      history,
    });

    expect(component).toMatchSnapshot();
  });
});
