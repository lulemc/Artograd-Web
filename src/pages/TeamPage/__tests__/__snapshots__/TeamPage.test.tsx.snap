// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team page Should create the team page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        Team Page
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Team Page
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
